File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/WordDocument.java
class WordDocument extends Document {
    public WordDocument(String name) {
        super(name);
    }
    @Override
    void open() {
        System.out.println("Opening Word document: " + name);
    }
    @Override
    void save() {
        System.out.println("Saving docs " + name);
    }
}

File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/WordDocument.cs
public class WordDocument : Document
{
    public WordDocument(string name) : base(name) {}

    public override void Open()
    {
        Console.WriteLine("Opening Word document: " + name);
    }
    public override void Save()
    {
        Console.WriteLine("Saving docs " + name);
    }
}
